src/store/modules/data/checkout.js

```diff
@@ -27,20 +26,39 @@ export const fetchCheckoutDataLoading = createAction(FETCH_CHECKOUT_DATA_LOADING
 // Thunks
 // ------------------------------------
 
-export function getCheckout () {
+export function getCheckout (poprMultihash) {
+  console.log(poprMultihash)
   return async dispatch => {
     dispatch(fetchCheckoutDataLoading(true))
+    let popr = null
     try {
-      // TODO: replace fixture data with stuff from PoPR
-      // TODO: handle error in UI
-      const vendorId = process.env.REACT_APP_VENDOR_ID || 'Qmtest'
-      const url = process.env.REACT_APP_MARKETPLACE_URL || 'http://localhost:4000'
-      const popr = await requestPopr(url, vendorId, {
-        // TODO: fix the placeholder price
-        amount: checkoutData.price * 10000,
-        currency_symbol: 'tBTC'
-      })
-      dispatch(fetchCheckoutDataSuccess(Object.assign({}, popr, checkoutData)))
+      if (poprMultihash) {
+        console.log('Reading PoPR at ' + poprMultihash)
+        const chluIpfs = await getChluIPFS()
```

Comment:
I don't know much about react, but will this cause the UI to block?

Reply:
Absolutely not :+1: